File: wun-android/src/main/kotlin/wun/foundation/HeadingRenderer.kt
// Compose renderer for `:wun/Heading`. `:level` 1-4 maps to a font
// scale; defaults to 2 (matching the brief's :wun/Text :variant :h1
// / :h2, but as the explicit primitive in the vocabulary).

package wun.foundation

import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import kotlinx.serialization.json.intOrNull
import kotlinx.serialization.json.jsonPrimitive
import wun.WunComponent

object HeadingRenderer {
    val render: WunComponent = { props, children ->
        val level = (props["level"]?.jsonPrimitive?.intOrNull) ?: 2
        val (size, weight) = when (level) {
            1    -> 30.sp to FontWeight.Bold
            2    -> 22.sp to FontWeight.SemiBold
            3    -> 18.sp to FontWeight.SemiBold
            else -> 16.sp to FontWeight.SemiBold
        }
        Text(text = Children.flatten(children),
             fontSize = size,
             fontWeight = weight)
    }
}

File: wun-ios/Sources/Wun/Foundation/HeadingRenderer.swift
// SwiftUI renderer for `:wun/Heading`. Maps `:level` 1-4 to SwiftUI
// font roles. `:wun/Text {:variant :h1}` already covered the heading-1
// case for the spike; `:wun/Heading` is the explicit primitive,
// matching the brief's vocabulary.

import SwiftUI

public enum WunHeading {
    public static let render: WunComponent = { props, children in
        let level = props["level"]?.intValue ?? 2
        let label = WunChildren.flatten(children)
        let font: Font = {
            switch level {
            case 1: return .largeTitle.weight(.bold)
            case 2: return .title.weight(.semibold)
            case 3: return .title2.weight(.semibold)
            default: return .title3.weight(.semibold)
            }
        }()
        return AnyView(SwiftUI.Text(label).font(font))
    }
}
